ibis/backends/flink/compiler/core.py

```diff
@@ -46,15 +49,53 @@ class FlinkCompiler(Compiler):
 
 def translate(op: ops.TableNode) -> str:
     # TODO(chloeh13q): support translation of non-select exprs (e.g. literals)
-    ast = FlinkCompiler.to_ast(op)
-    return translate_language(ast.queries[0])
+    return translate_op(op)
 
 
 @functools.singledispatch
-def translate_language(language: DML | DDL) -> str:
-    raise com.OperationNotDefinedError(f'No translation rule for {type(language)}')
-
-
-@translate_language.register(Select)
-def _select(language: Select):
-    return language.compile()
+def translate_op(op: ops.TableNode) -> str:
+    raise com.OperationNotDefinedError(f'No translation rule for {type(op)}')
+
+
+@translate_op.register(ops.Literal)
+def _literal(op: ops.Literal) -> str:
+    value = op.value
+    dtype = op.output_dtype
+
+    if dtype.is_boolean():
+        # TODO(chloeh13q): Flink supports a third boolean called "UNKNOWN"
+        return 'TRUE' if value else 'FALSE'
+    elif dtype.is_string():
+        quoted = value.replace("'", "''").replace("\\", "\\\\")
+        return f"'{quoted}'"
+    elif dtype.is_date():
+        if isinstance(value, datetime.date):
+            value = value.strftime('%Y-%m-%d')
+        return repr(value)
+    elif dtype.is_numeric():
+        if math.isnan(value):
+            raise ValueError("NaN is not supported in Flink SQL")
```

Comment:
Can it not be translated to null?

Reply:
I've discussed this with @zhenzhongxu - I'm not sure if it's entirely appropriate / accurate to translate NaN into NULL because they have different meanings (also see [Flink's documentation on NULL](https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/table/types/#null)). We were not able to find any documentation on whether `NaN` exists in Flink, so I think it might be best to error out for now.